File: src/main/java/org/taktik/icure/entities/embed/Periodicity.java
package org.taktik.icure.entities.embed;

import org.taktik.icure.entities.base.CodeStub;
import org.taktik.icure.validation.AutoFix;
import org.taktik.icure.validation.ValidCode;

import java.io.Serializable;

public class Periodicity implements Serializable {
    @ValidCode(autoFix = AutoFix.NORMALIZECODE)
    protected CodeStub relatedCode;
    @ValidCode(autoFix = AutoFix.NORMALIZECODE)
    protected CodeStub relatedPeriodicity;

    public CodeStub getRelatedCode() { return relatedCode; }

    public void setRelatedCode(CodeStub relatedCode) { this.relatedCode = relatedCode; }

    public CodeStub getRelatedPeriodicity() { return relatedPeriodicity; }

    public void setRelatedPeriodicity(CodeStub relatedPeriodicity) { this.relatedPeriodicity = relatedPeriodicity; }
}

File: src/main/kotlin/org/taktik/icure/entities/embed/Periodicity.kt
package org.taktik.icure.entities.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import org.taktik.icure.entities.base.CodeStub
import org.taktik.icure.validation.AutoFix
import org.taktik.icure.validation.ValidCode
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class Periodicity(
        @field:ValidCode(autoFix = AutoFix.NORMALIZECODE) val relatedCode: CodeStub? = null,
        @field:ValidCode(autoFix = AutoFix.NORMALIZECODE) val relatedPeriodicity: CodeStub? = null
) : Serializable
